### ISSUE TITLE
self.config is deprecated

### ISSUE THREAD

**Issue by forslund** [forslund/cmd_skill#2]
self.config is deprecated, the skill should offer usage through skill settings.

**Comment by AlphaPy** [comment]
I've done this:
`self.settings["uid"] = None
        self.settings["gid"] = None
        self.settings["alias"] = {}
        if self.settings:
            user = self.settings["uid"]
            if user:
                pwnam = getpwnam(user)
                self.settings["uid"] = pwnam.pw_uid
                self.settings["gid"] = pwnam.pw_gid
            self.settings["alias"] = self.config.get('alias', {})`
But I haven't fixed that last line yet.

**Comment by forslund** [comment]
Hmm... I was thinking something along the lines of

in the initialize having something like:
```python

user = self.settings.get('user') or self.config.get('user')
if user:
    pwnam = getpwnam(user)
    self.uid = pwnam.pw_uid
    self.gid = pwnam.pw_gid
self.alias = self.settings.get('alias') or self.config.get('alias') or {}
```

Which should work well for a transition period. Reading the settings.json for the skill if they exist otherwise falling back to the old config.

What do you think

**Comment by AlphaPy** [comment]
I apologize for deleting my comment. My understanding of Python at the moment is very basic as I have been trying to pick it up in my spare time. I made some errors in the code and deleted them. I managed to change the deprecated code without errors but it no longer functions. I'm still tinkering with it but I like your idea. Thanks again for this awesome skill!
Edit: I think I understand now what I did wrong but I'm too tired to press on tonight. I'll take another shot at it tomorrow if I have time.
